### ISSUE TITLE
streaming application should be run before data generation

### ISSUE THREAD

**Issue by manuzhang** [intel-hadoop/HiBench#389]
In [run-streamingbench.md](https://github.com/intel-hadoop/HiBench/blob/master/docs/run-streamingbench.md), users are instructed to launch spark streaming application after running `dataGen.sh`. This will result in **inaccurate and  higher** latency since `latency = latency1(DataGenToKafka) + latency2(KafkaToStreamingApplication) + latency3(StreamingApplication)` and `latency2` also includes the time to launch streaming application. Instead, streaming application should be launched and run **before** `dataGen.sh` such that `latency2` only measures how long it takes to consume from kafka.

**Comment by carsonwang** [comment]
Thanks @manuzhang. This makes sense. Do you mind create a PR to fix the doc? By the way, will the topic be created automatically if we launch the streaming app first?

**Comment by manuzhang** [comment]
No, only kafka producer will create topic automatically. We can move [MetricsUtil#createTopic](https://github.com/intel-hadoop/HiBench/blob/master/common/src/main/scala/com/intel/hibench/common/streaming/metrics/MetricsUtil.scala#L36) to a more common place and reuse it when launching the app.
